packages/teleport-generator-shared/src/builders/uidl-builders.ts

```diff
@@ -19,24 +20,30 @@ export const component = (name: string, node: UIDLNode): ComponentUIDL => {
 export const elementNode = (
   elementType: string,
   attrs?: Record<string, UIDLAttributeValue>,
-  children?: UIDLNode[]
+  children?: UIDLNode[],
+  style?: Record<string, UIDLStyleValue>
 ): UIDLElementNode => {
   return {
     type: 'element',
-    content: element(elementType, attrs, children),
+    content: style
+      ? element(elementType, attrs, children, style)
+      : element(elementType, attrs, children),
   }
 }
 
 export const element = (
   elementType: string,
   attrs?: Record<string, UIDLAttributeValue>,
-  children?: UIDLNode[]
+  children?: UIDLNode[],
+  style?: Record<string, UIDLStyleValue>
 ) => {
   return {
     elementType,
     name: elementType,
     attrs,
+    style,
     children,
+    key: elementType,
```

Comment:
why is the `key` configured here? the resolver should generate unique keys for the entire component

Reply:
if it's needed specific for a test to generate the className properly, then it should be added on the spot when setting up that specific test. Otherwise, this function should remain generic as the standard behavior when building a UIDL is to NOT have the `key` attribute and let the resolver handle it